sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -996,9 +1076,12 @@ public void populateDisplayData(DisplayData.Builder builder) {
 
       builder
           .addIfNotNull(DisplayData.item("table", getJsonTableRef())
-            .withLabel("Table Reference"))
-          .addIfNotNull(DisplayData.item("schema", getJsonSchema())
-            .withLabel("Table Schema"));
+            .withLabel("Table Reference"));
+      if (getSchemaFunction() instanceof ConstantSchemaFunction) {
+        ValueProvider<String> jsonSchema = ((ConstantSchemaFunction) getSchemaFunction())
+            .jsonSchema;
+        builder.addIfNotNull(DisplayData.item("schema", jsonSchema).withLabel("Table Schema"));
```

Comment:
Maybe add a DisplayData item specifying that this is using a custom schema function otherwise?